js_modules/dagit/packages/core/src/workspace/VirtualizedGraphTable.tsx

```diff
@@ -0,0 +1,144 @@
+import {gql, useLazyQuery} from '@apollo/client';
+import {Box, Caption, Colors} from '@dagster-io/ui';
+import {useVirtualizer} from '@tanstack/react-virtual';
+import * as React from 'react';
+import {Link} from 'react-router-dom';
+import styled from 'styled-components/macro';
+
+import {Container, HeaderCell, Inner, Row, RowCell} from '../ui/VirtualizedTable';
+
+import {useDelayedRowQuery} from './VirtualizedWorkspaceTable';
+import {RepoAddress} from './types';
+import {SingleGraphQuery, SingleGraphQueryVariables} from './types/SingleGraphQuery';
+import {workspacePathFromAddress} from './workspacePath';
+
+export type Graph = {name: string; path: string; description: string | null};
+
+interface Props {
+  graphs: Graph[];
+  repoAddress: RepoAddress;
+}
+
+export const VirtualizedGraphTable: React.FC<Props> = ({repoAddress, graphs}) => {
+  const parentRef = React.useRef<HTMLDivElement | null>(null);
+
+  const rowVirtualizer = useVirtualizer({
+    count: graphs.length,
+    getScrollElement: () => parentRef.current,
+    estimateSize: () => 64,
+    overscan: 10,
+  });
+
+  const totalHeight = rowVirtualizer.getTotalSize();
+  const items = rowVirtualizer.getVirtualItems();
+
+  return (
+    <>
+      <Box
+        border={{side: 'horizontal', width: 1, color: Colors.KeylineGray}}
+        style={{
+          display: 'grid',
+          gridTemplateColumns: '100%',
+          height: '32px',
+          fontSize: '12px',
+          color: Colors.Gray600,
+        }}
+      >
+        <HeaderCell>Graph</HeaderCell>
+      </Box>
+      <div style={{overflow: 'hidden'}}>
```

Comment:
Hmm what impact does this `<div>` have?  I didn't think hidden would do anything on it's own -- does this do something similar to a `min-width:0`?

Reply:
This restricts the height of the virtualized table so that we don't end up scrolling past it.